internal/registry/registry.go

```diff
@@ -8,45 +8,60 @@ import (
 	cloudevents "github.com/cloudevents/sdk-go/v2"
 )
 
-// A declaratively registered function
+// RegisteredFunction represents a function that has been
+// registered with the registry.
 type RegisteredFunction struct {
 	Name         string                                         // The name of the function
 	CloudEventFn func(context.Context, cloudevents.Event) error // Optional: The user's CloudEvent function
 	HTTPFn       func(http.ResponseWriter, *http.Request)       // Optional: The user's HTTP function
 }
 
-var (
-	function_registry = map[string]RegisteredFunction{}
-)
+// Registry is a registry of functions.
+type Registry struct {
+	functions map[string]RegisteredFunction
+}
+
+var defaultInstance = New()
+
+// Default returns the default, singleton registry instance.
+func Default() *Registry {
+	return defaultInstance
+}
+
+func New() *Registry {
+	return &Registry{
+		functions: map[string]RegisteredFunction{},
+	}
+}
```

Comment:
I don't think we want to support multiple registries of functions. Can we just have one global registry instance?

Reply:
We won't support multiple registries in functions framework, but this is better software engineering practice typically. This way other packages could theoretically reuse the registry package for their own uses without messing with the Default registry.

Also, it helps with the tests. The tests were technically stomping over each other, all messing with the global registry without clearing it. Now, they are proper unit tests only testing their own behavior.

Reply:
Yeah, I could see both ways. It should be easier for testing as you mentioned.